### PR TITLE
add fromKeyPair

### PR DESCRIPTION
**Pull request by mattlockyer**
for #450

### CODE REVIEW

File: src/signer.ts [hunk 1]
```diff
@@ -12,6 +12,14 @@ export abstract class Signer {
      */
     abstract async createKey(accountId: string, networkId?: string): Promise<PublicKey>;
 
+    /**
+     * Creates a public key for the account, network and keyPair provided
+     * @param accountId accountId to retrieve from.
+     * @param networkId The targeted network. (ex. default, betanet, etc…)
+     * @param keyPair The keyPair to use for signing
+     */
+    abstract async fromKeyPair(accountId: string, networkId: string, keyPair: KeyPair): Promise<PublicKey>;
```
Comment (vgrichina) [inline src/signer.ts:21]: @mattlockyer let's keep the same order of `networkId` / `accountId` as in `setKey` to avoid additional confusion

**Comment by mattlockyer** [comment]
Not sure why pr build isn't working? Ran tests locally all passed.

File: src/signer.ts [hunk 2]
```diff
@@ -50,6 +58,18 @@ export class InMemorySigner extends Signer {
...
     }
+    
+    /**
+     * Creates a public key for the account, network and keyPair provided
+     * @param networkId The targeted network. (ex. default, betanet, etc…)
+     * @param accountId The NEAR account to assign a public key to
+     * @param keyPair The keyPair to use for signing
+     * @returns {Promise<PublicKey>}
+     */
+    async fromKeyPair(networkId: string, accountId: string, keyPair: KeyPair): Promise<PublicKey> {
```
Comment (vgrichina) [inline src/signer.ts:69]: @mattlockyer sorry that missed it the first time, but I think following API makes most sense:

`InMemorySigner.fromKeyPair(networkId: string, accountId: string, keyPair: KeyPair): Promise<Signer>`

This is useful every time we need to have throwaway signer. 

Basically create signer instance as a whole from single KeyPair. Generally this only makes sense for in-memory signer, i.e. it won't work for Ledger, etc.

**Comment by vgrichina** [comment]
> Not sure why pr build isn't working? Ran tests locally all passed.

Looks like a flake

**Pull request merged**